app/views/documents/index/_filters.html.erb

```diff
@@ -0,0 +1,43 @@
+<%= form_tag({}, method: :get) do %>
+  <%= render "govuk_publishing_components/components/input", {
+    label: {
+      text: t("documents.index.filter.title_or_url")
+    },
+    name: "title_or_url",
+    value: params[:title_or_url],
+    type: "search"
+  } %>
+
+  <% document_type_select = SupertypeSchema.all.inject([]) do |memo, supertype|
```

Comment:
Perhaps this can be extracted to the `DocumentFilter` class?

Reply:
As it's very view-specific - generating an array of arrays - I thought it would be best here, a la choose_supertype.html.erb.

Reply:
Ok, good point. Could you golf it down to this?

```rb
DocumentTypeSchema.all.reject(&:managed_elsewhere).map { |d| [d.label, d.id] }
```

Reply:
Done - 'twas an unnecessary relic.